### ISSUE TITLE
Nextcloud 15 Beta 2 | Wrong app update availability check on update

### ISSUE THREAD

**Issue by MichaIng** [nextcloud/server#12477]
To make long story short, the v15 compatible app list seems to be simply not online (yet). Check browser log at the bottom: `GET https://my.domain,org/nextcloud/ocs/v2.php/apps/updatenotification/api/v1/applist/Nextcloud%2015.0.0%20beta%202 404 (Not Found)`

### Steps to reproduce
1. Install Nextcloud 14.0.3 (current latest stable)
2. Configure update channel "beta"
3. See available (currently) Nextcloud 15 Beta 2 update with the info that all apps can be updated as well.
4. Run the update and see, that ~half of the apps couldn't be updated 😮. 

### Expected behaviour
Admin panel should correctly inform about apps that cannot be updated to a NC15 compatible version.

### Actual behaviour
It shows info that all apps can be updated to a NC15 compatible version.

### Server configuration

**Operating system**: Raspbian Buster | Debian Stretch

**Web server:** Apache 2.4.37 |  Lighttpd 1.4.45

**Database:** MariaDB 10.1.37 | MariaDB 10.1.26

**PHP version:** Mod_PHP 7.2 | PHP-FPM 7.0.30

**Nextcloud version:** 14.0.3 (stable) => 15.0.0 Beta 2

**Updated from an older Nextcloud/ownCloud or fresh install:** older | fresh

**Where did you install Nextcloud from:** https://download.nextcloud.com/server/

**Signing status:** `No errors have been found.`

**List of activated apps:** Now not much, also obviously irrelevant, checked on two systems, one fresh with default apps only

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "my.domain.org"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "15.0.0.6",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": true,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "loglevel": 1,
        "logtimezone": "Europe\/Berlin",
        "trashbin_retention_obligation": "disabled",
        "versions_retention_obligation": "disabled",
        "skeletondirectory": "",
        "defaultapp": "apporder",
        "maintenance": false,
        "overwrite.cli.url": "https:\/\/localhost\/nextcloud",
        "htaccess.RewriteBase": "\/nextcloud",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpsecure": "ssl",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "mysql.utf8mb4": true,
        "updater.release.channel": "beta",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Opera dev 58.0.3135.3

**Operating system:** Windows 10 Pro N 1809

#### Browser log
<details>
<summary>Browser log</summary>

**GET https://my.domain.org/nextcloud/ocs/v2.php/apps/updatenotification/api/v1/applist/Nextcloud%2015.0.0%20beta%202 404 (Not Found)**

send | @ | core.js?v=6aeb4cdf-0:4
---- | - | ----------------------
ajax | @ | core.js?v=6aeb4cdf-0:4
isNewVersionAvailable | @ | root.vue:155
jt.run | @ | vue.esm.js:3233
Lt | @ | vue.esm.js:2981
(anonymous) | @ | vue.esm.js:1837
Xe | @ | vue.esm.js:1758
Promise.then (async) |   |  
Ke | @ | vue.esm.js:1806
tt | @ | vue.esm.js:1850
(anonymous) | @ | vue.esm.js:3068
jt.update | @ | vue.esm.js:3209
le.notify | @ | vue.esm.js:697
set | @ | vue.esm.js:1014
Nt.Pt.set | @ | vue.esm.js:3300
beforeMount | @ | root.vue:318
Ot | @ | vue.esm.js:2921
(anonymous) | @ | vue.esm.js:2765
dn.$mount | @ | vue.esm.js:8540
dn.$mount | @ | vue.esm.js:10939
init | @ | vue.esm.js:4137
(anonymous) | @ | vue.esm.js:5608
f | @ | vue.esm.js:5555
(anonymous) | @ | vue.esm.js:6130
e._update | @ | vue.esm.js:2660
(anonymous) | @ | vue.esm.js:2788
jt.get | @ | vue.esm.js:3142
jt | @ | vue.esm.js:3131
(anonymous) | @ | vue.esm.js:2795
dn.$mount | @ | vue.esm.js:8540
dn.$mount | @ | vue.esm.js:10939
(anonymous) | @ | init.js:38
n | @ | bootstrap:19
(anonymous) | @ | bootstrap:83
(anonymous) | @ | bootstrap:83
</details>

**Comment by MorrisJobke** [comment]
This is fixed with 14.0.4 RC 1 and 15.0.0 beta 2. Unfortunately will only show from now on. 

Fix is in #12358

**Comment by MorrisJobke** [comment]
We added a note to the blog post about this: https://nextcloud.com/blog/keep-it-up-beta-2-is-here/

Workaround is to upgrade to 14.0.4 RC 1 first to get the fix.
